apps/swirl-docs/src/pages/api-docs/[apiSpec].tsx

```diff
@@ -0,0 +1,291 @@
+import { ApiDoc, Endpoint, createStaticPathsForSpecs } from "@swirl/lib/docs";
+import Head from "next/head";
+import { DocumentationLayout } from "../../components/Layout/DocumentationLayout";
+import { GetStaticPaths, GetStaticProps } from "next";
+import { ScriptProps } from "next/script";
+import { apiDocsNavItems } from "@swirl/lib/navigation/src/data/apiDocs.data";
+import OASBuilder from "@swirl/lib/docs/src/OasBuilder";
+import OASNormalize from "oas-normalize";
+import { serializeMarkdownString } from "@swirl/lib/docs/src/singleDoc";
+import { MDXRemoteSerializeResult } from "next-mdx-remote";
+import { useEffect, useState } from "react";
+import { ReactMarkdown } from "react-markdown/lib/react-markdown";
+import { CodePreview } from "src/components/CodePreview";
+import { Tag } from "src/components/Tags";
+import { Parameter } from "src/components/Documentation/Parameter";
+import { SwirlIconOpenInNew } from "@getflip/swirl-components-react";
+
+async function getSpecData(spec: string): Promise<ApiDoc> {
+  const navItem = apiDocsNavItems.find((item) => item.url.includes(spec));
+  const specPath = navItem?.specPath;
+
+  const oasDocument = await new OASNormalize(specPath, {
+    enablePaths: true,
+  }).validate();
+
+  const oasBuilder = await new OASBuilder(oasDocument);
+
+  oasBuilder.setTitleAndPath().setDescription().setEndpoints().setOperations();
+
+  return {
+    title: oasBuilder.title,
+    path: oasBuilder.path,
+    definition: oasBuilder.oasDocument,
+    shortDescription: oasBuilder.shortDescription,
+  };
+}
+
+export const getStaticPaths: GetStaticPaths = async () => {
+  const specs = createStaticPathsForSpecs();
+
+  return {
+    fallback: false,
+    paths: specs,
+  };
+};
+
+function generateGeneralDescription(document: ApiDoc) {
+  return document.definition?.info.description
+    ?.replace(document.title, "")
+    .replace(document.shortDescription, "")
```

Comment:
headsup: i know this only works for now, but i don't want to make the PR bigger as it is to add a module to handle replacing and handling the strings in the markdown string for now. i hope this is ok 😅